test/extensions/tracers/opencensus/config_test.cc

```diff
@@ -322,6 +322,9 @@ TEST(OpenCensusTracerConfigTest, OpenCensusHttpTracerStackdriverGrpc) {
         google_grpc:
           target_uri: 127.0.0.1:55678
           stat_prefix: test
+        initial_metadata:
+        - key: foo
+          value: bar
```

Comment:
Looks much nicer! Is there a way to verify that the options have been set in the OpenCensus registration?

Reply:
I don't think so, It does not expose the client context or the stub.

Reply:
OK, so this isn't any worse than the existing state of these tests. Before merging I'd like to check with @g-easy if there is a way or a plan to make this more testable that we can track.

Reply:
I'm sorry, I'm very limited in the time I can spend on OpenCensus. I can review PRs. How would you suggest improving testability here? In many cases, we limited our API on purpose but we can open it more where necessary.

Reply:
@g-easy not asking for any development work, just advice on how we could construct an issue to track tech debt around testing? I.e. if someone had cycles, how would you advise they go about making options like this testable?

Reply:
For something like this, could we pass a mock stub to the Stackdriver trace exporter and expect that the extra metadata is included in requests?

There's already an option for passing custom stubs in the trace exporter options.

Reply:
Yeah, that is feasible, but would require generating mock stub interface for stackdriver tracing service, and I am not sure whether Envoy would like to take that complexity or not.

Reply:
@bianpengyuan is this just a simple gmock class? Or are you asking to engineer a fake object with some complexity? We'd definitely take the former.

Reply:
just a gmock class, like https://grpc.github.io/grpc/cpp/md_doc_unit_testing.html

Reply:
If that sounds good, I can add a TODO here to track testing for this.

Reply:
Yes, thanks.

Reply:
Some time ago I persuaded the googleapis people to enable generate_mocks e.g. https://github.com/googleapis/googleapis/blob/5c8ab2c072d557c2f4c4e54b544394e2d62202d5/google/devtools/cloudtrace/v2/BUILD.bazel#L152